api/types/desktop.go

```diff
@@ -127,6 +127,198 @@ func (s *WindowsDesktopServiceV3) MatchSearch(values []string) bool {
 	return MatchSearch(fieldVals, values, nil)
 }
 
+// DynamicWindowsDesktop represents a Windows desktop host that is automatically discovered by Windows Desktop Service.
+type DynamicWindowsDesktop interface {
+	// ResourceWithLabels provides common resource methods.
+	ResourceWithLabels
+	// GetAddr returns the network address of this host.
+	GetAddr() string
+	// GetDomain returns the ActiveDirectory domain of this host.
+	GetDomain() string
+	// NonAD checks whether this is a standalone host that
+	// is not joined to an Active Directory domain.
+	NonAD() bool
+	// GetScreenSize returns the desired size of the screen to use for sessions
+	// to this host. Returns (0, 0) if no screen size is set, which means to
+	// use the size passed by the client over TDP.
+	GetScreenSize() (width, height uint32)
+	// Copy returns a copy of this windows desktop
+	Copy() *DynamicWindowsDesktopV1
+	// CloneResource returns a copy of the WindowDesktop as a ResourceWithLabels
+	CloneResource() ResourceWithLabels
+}
+
+var _ DynamicWindowsDesktop = &DynamicWindowsDesktopV1{}
+
+// NewDynamicWindowsDesktopV1 creates a new DynamicWindowsDesktopV1 resource.
+func NewDynamicWindowsDesktopV1(name string, labels map[string]string, spec DynamicWindowsDesktopSpecV1) (*DynamicWindowsDesktopV1, error) {
+	d := &DynamicWindowsDesktopV1{
+		ResourceHeader: ResourceHeader{
+			Metadata: Metadata{
+				Name:   name,
+				Labels: labels,
+			},
+		},
+		Spec: spec,
+	}
+	if err := d.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return d, nil
+}
+
+func (d *DynamicWindowsDesktopV1) setStaticFields() {
+	d.Kind = KindDynamicWindowsDesktop
+	d.Version = V1
+}
+
+// CheckAndSetDefaults checks and sets default values for any missing fields.
+func (d *DynamicWindowsDesktopV1) CheckAndSetDefaults() error {
+	if d.Spec.Addr == "" {
```

Comment:
Should we also verify is the address makes sense?

Reply:
I don't think it makes sense here, I'll leave it as is